socialhome/frontend/routes.js

```diff
@@ -0,0 +1,25 @@
+import VueRouter from "vue-router"
+import Stream from "frontend/components/streams/Stream.vue"
+
+
+function $$(props = {}) {
+    return route => Object.assign({}, route.params, props)
+}
+
+const routes = [
+    // TODO: Matching '/' is a bit tricky since the stream can be customized.
+    // TODO: In the future, Stream can be replaced by a generic RootStream.vue that just delegates to another component
+    {path: "/", component: Stream},
+    {path: "/streams/followed", component: Stream},
+    {path: "/streams/public", component: Stream},
+    {path: "/u/:user", component: Stream, props: $$()},
+    {path: "/u/:user/all", component: Stream, props: $$()},
+    {path: "/p/:guid", component: Stream, props: $$()},
+    {path: "/p/:guid/all", component: Stream, props: $$()},
+    {path: "/streams/tag/:name", component: Stream, props: $$()},
+]
```

Comment:
Single content view (stream!) is missing, basically `/content/<id>/<shorttext>/`.

The short text is just eye candy (nicer urls), you can ignore it for routing, only the ID matters. Sometimes there is no short text at all for weird languages that can't be slugified, so you'll have to support also `/content/<id>/` style version.

Oh another comment, all the urls in Django have a postfix `/` which is missing from all the urls above. Maybe it doesn't matter here but would like to keep this style for consistency.